chapters/en/slides/module5/module5_03.Rmd

```diff
@@ -0,0 +1,316 @@
+
+---
+
+## Read in the data containing the daily sum of donations to Wikipedia in 2020
+
+```{python}
+# TODO save wrangled data?
+import pandas as pd
+import altair as alt
+from vega_datasets import data
+
+donations = (
+    pd.read_csv("https://frdata.wikimedia.org/donationdata-vs-day.csv", parse_dates=['date'])
+    .query('2020 < date < 2021')
+    .assign(week_day=lambda df: df['date'].dt.day_name().str[:3]))
+
+alt.Chart(donations, title='Wikipedia donations peak during Christmas').mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+We will be working with a data set that contains the daily sum
+of all donations received by Wikipedia
+in the year 2020.
+
+From this overview of the data
+we can see that the most money
+is donated around Christmas and new years.
+
+There also seems to be a higher amount donated during the fall
+than during spring and winter.
+
+However,
+we cannot see exactly which values the data points during the first half of the year has.
+Are they close to zero? Or 10,000? We can't know from looking at this plot.
+
+---
+
+## We can zoom in to highlight a region of an axes by filtering the data
+
+```{python}
+title_text = 'Wikipedia donations peak during Christmas'
+alt.Chart(donations[donations['date'].between("2020", "2020-07")], title=title_text).mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+Zooming in on an axis is useful
+when we want to highlight a certain region.
+The easiest way to achieve this
+is usually to filter the data we pass to `Chart` as in this example.
+
+---
+
+## We can zoom in to highlight a region of an axes
+
+```{python}
+title_text = 'Wikipedia donations peak during Christmas'
+alt.Chart(donations[donations['date'].between("2020", "2020-07")], title=title_text).mark_circle(clip=True).encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day', scale=alt.Scale(domain=[0, 100_000])))
+```
+
+Notes:
+It is not always practical to filter the data,
+so it is good to know how to change the axis range directly in Altair.
+For this,
+we could modify the domain of the axis' scale.
+
+When we used `alt.Axis` previously,
+the changes we made were cosmetic,
+such as tick numbers,
+label formatting etc.
+
+Now that were are modifying the range of the actual displayed values,
+we need to use `alt.Scale` instead.
+It also works for colors and sizes in a legend,
+just like `alt.Axis`.
+
+Finally we need to set `clip=True` in the axis,
+to be explicit about that we are excluding data.
+Without it,
+we would still see the circles extending beyond the range of the chart.
+
+This ensures that we are intentional about our data exclusion,
+and it is something we should think twice about because it can skew the perception of the data.
```

Comment:
Is this really why? Or is it because altair is super flexible, and maybe there are times you might want to have points show up that are beyond the chart area, for whatever reason.